src/python_routing_v02/troute/routing/fast_reach/simple_da.pyx

```diff
@@ -51,6 +50,11 @@ cdef (float, float, float, float) simple_da(
         # add/update lastobs_timestep
         lastobs_time = (timestep) * routing_period
         lastobs_val = target_val
+    if ((isnan(target_val)) and (isnan(lastobs_val))):
+        replacement_val = model_val
+        nudge_val = 0.0
+        lastobs_val = NAN
+        lastobs_time = NAN
```

Comment:
@jameshalgren can you confirm that the order of these if statements followed by the else is correct?

Reply:
It was not correct... I will push a suggested correction.